hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -51,15 +52,135 @@ sealed abstract class Block extends Product with AutoLocated:
     case Define(_, rst) => 1 + rst.size
     case TryBlock(sub, fin, rst) => 1 + sub.size + fin.size + rst.size
     case Label(_, bod, rst) => 1 + bod.size + rst.size
+    case HandleBlock(lhs, res, par, cls, handlers, bdy, rst) => 1 + handlers.map(_.body.size).sum + bdy.size + rst.size
+  
+  // ignoring blocks inside functions and handle block
+  def map(f: Block => Block): Block = this match
+    case _: Return | _: Throw | _: End | _: Break | _: Continue | _: HandleBlockReturn => this
+    case Match(scrut, arms, dflt, rst) => Match(scrut, arms.map(_ -> f(_)), dflt.map(f), f(rst))
+    case Label(lbl, bod, rst) => Label(lbl, f(bod), f(rst))
+    case Begin(sub, rst) => Begin(f(sub), f(rst))
+    case TryBlock(sub, fin, rst) => TryBlock(f(sub), f(fin), f(rst))
+    case Assign(l, r, rst) => Assign(l, r, f(rst))
+    case b @ AssignField(l, n, r, rst) => AssignField(l, n, r, f(rst))(b.symbol)
+    case Define(defn, rst) => Define(defn, f(rst))
+    case HandleBlock(l, res, par, cls, hdr, bod, rst) => HandleBlock(l, res, par, cls, hdr, bod, f(rst))
+  
+  def mapResult(f: Result => Opt[(Result => Block) => Block]): Block = this match
+    case Return(res, implct) => f(res).map(_(Return(_, implct))).getOrElse(this)
+    case Throw(exc) => f(exc).map(_(Throw(_))).getOrElse(this)
+    case Assign(l, r, rst) => f(r).map(_(Assign(l, _, rst))).getOrElse(this)
+    case b @ AssignField(l, n, r, rst) => f(r).map(_(AssignField(l, n, _, rst)(b.symbol))).getOrElse(this)
+    case HandleBlockReturn(res) => f(res).map(_(HandleBlockReturn(_))).getOrElse(this)
+    case _: End | _: Break | _: Continue | _: Match | _: Label | _: Begin | _: TryBlock | _: Define | _: HandleBlock | _: HandleBlockReturn => this
+  
+  def mapPath(f: Path => Path): Block = this match
+    case Return(res: Path, implct) => Return(f(res), implct)
+    case Throw(exc: Path) => Throw(f(exc))
+    case Assign(l, r: Path, rst) => Assign(l, f(r), rst)
+    case b @ AssignField(l, n, r: Path, rst) => AssignField(l, n, f(r), rst)(b.symbol)
+    case Match(scrut: Path, arms, dflt, rst) => Match(f(scrut), arms, dflt, rst)
+    case Define(ValDefn(owner, k, sym, rhs: Path), rst) => Define(ValDefn(owner, k, sym, f(rhs)), rst)
+    case HandleBlockReturn(res: Path) => HandleBlockReturn(f(res))
+    case _: Return | _: Throw | _: Assign | _: AssignField | _: End | _: Break | _: Continue | _: Match | _: Label | _: Begin | _: TryBlock | _: Define | _: HandleBlock | _: HandleBlockReturn => this.mapResult {
```

Comment:
Please try not to exceed 100-character lines.

Also braces are lame:
```suggestion
    case _: Return | _: Throw | _: Assign | _: AssignField | _: End | _: Break | _: Continue | _: Match | _: Label | _: Begin | _: TryBlock | _: Define | _: HandleBlock | _: HandleBlockReturn => this.mapResult:
```